src/sinks/elasticsearch/common.rs

```diff
@@ -133,37 +135,45 @@ impl ElasticsearchCommon {
         })
     }
 
-    pub fn signed_request(&self, method: &str, uri: &Uri, use_params: bool) -> SignedRequest {
-        let mut request = SignedRequest::new(method, "es", &self.region, uri.path());
-        request.set_hostname(uri.host().map(|host| host.into()));
-        if use_params {
-            for (key, value) in &self.query_params {
-                request.add_param(key, value);
-            }
-        }
-        request
-    }
-
     pub async fn healthcheck(self, client: HttpClient) -> crate::Result<()> {
         let mut builder = Request::get(format!("{}/_cluster/health", self.base_url));
 
-        match &self.credentials {
-            None => {
-                if let Some(authorization) = &self.authorization {
-                    builder = authorization.apply_builder(builder);
-                }
-            }
-            Some(credentials_provider) => {
-                let mut signer = self.signed_request("GET", builder.uri_ref().unwrap(), false);
-                builder = finish_signer(&mut signer, credentials_provider, builder).await?;
-            }
+        if let Some(authorization) = &self.http_auth {
+            builder = authorization.apply_builder(builder);
+        }
+        let mut request = builder.body(Bytes::new())?;
+
+        if let Some(credentials_provider) = &self.aws_auth {
+            sign_request(&mut request, credentials_provider, &self.region).await?;
         }
-        let request = builder.body(Body::empty())?;
-        let response = client.send(request).await?;
+        let response = client.send(request.map(hyper::Body::from)).await?;
 
         match response.status() {
             StatusCode::OK => Ok(()),
             status => Err(HealthcheckError::UnexpectedStatus { status }.into()),
         }
     }
 }
+
+pub async fn sign_request(
```

Comment:
This is something I would consider a bug in the `rusoto` crate, and is no longer an issue.

What was happening before was when we created a `SignedRequest` we passed in the `endpoint` and a region. Rusoto treated the url as a "path" instead of the full endpoint and when the signed request was generated, it created a new hostname based off of the region (ignoring the original hostname).

Rusoto creating the new hostname: https://github.com/rusoto/rusoto/blob/e9d921492ffb020d0d80aaf5438f3eb865941ee4/rusoto/signature/src/signature.rs#L532-L537

Rusoto building hostname based on region (if region is not explicitely set)
https://github.com/rusoto/rusoto/blob/e9d921492ffb020d0d80aaf5438f3eb865941ee4/rusoto/signature/src/signature.rs#L230-L237

The AWS SDK is more sane and keeps the original endpoint, so this is no longer an issue.

Reply:
Aha, I see 👍